go/libkb/util_test.go

```diff
@@ -151,3 +152,36 @@ func TestForceWallClock(t *testing.T) {
 	require.True(t, hasMonotonicClock(n))
 	require.False(t, hasMonotonicClock(ForceWallClock(n)))
 }
+
+func TestDecodeHexFixed(t *testing.T) {
+	units := []struct {
+		src string
+		dst []byte
+		err string
+	}{
+		{"", []byte{}, ""},
+		{"aa", []byte{170}, ""},
+		{"abcd", []byte{171, 205}, ""},
+
+		{"a", []byte{}, "encoding/hex: odd length hex string"},
+		{"aaa", []byte{170}, "encoding/hex: odd length hex string"},
+		{"aa", []byte{}, "error decoding fixed-length hex: expected 0 bytes but got 1"},
+		{"", []byte{170}, "error decoding fixed-length hex: expected 1 bytes but got 0"},
+		{"abcd", []byte{171, 205, 0}, "error decoding fixed-length hex: expected 3 bytes but got 2"},
+		{"abcd", []byte{171}, "error decoding fixed-length hex: expected 1 bytes but got 2"},
+	}
+	for i, unit := range units {
+		t.Logf("units[%v]", i)
+		buf := make([]byte, len(unit.dst))
+		err := DecodeHexFixed(buf, []byte(unit.src))
+		if unit.err != "" {
+			require.Error(t, err)
+			require.Equal(t, unit.err, err.Error())
```

Comment:
Since we're already using a concrete error type in this case, it might be nicer to put the expected/got numbers on it as members and check them, rather than duplicating the error string here?

Reply:
That sounds kind of elegant but this is easy and I don't think anyone will get bit